pkg/apis/hive/v1alpha1/clusterdeployment_types.go

```diff
@@ -72,6 +72,10 @@ type ClusterDeploymentSpec struct {
 	// +required
 	PullSecret corev1.LocalObjectReference `json:"pullSecret"`
 
+	// GlobalPullSecret should be used when additional secret needs to be given apart from PullSecret.
+	// +optional
+	GlobalPullSecret corev1.LocalObjectReference `json:"pullSecret"`
```

Comment:
GlobalPullSecret needs to be on HiveConfig, otherwise it would require the caller to set it on every ClusterDeployment. The goal here is for the cluster admin to be able to configure a pull secret to be used globally or in addition to whatever the user sets. (depending on the release image they use and what registry it's located on)

Reply:
HiveConfig is only read by the pkg/operator code, which in turn creates some environment variables to set on the hive-controllers Deployment. We want to transfer this secret name from HiveCOnfig, to an env var on the Deployment, in the our operators code.

Reply:
> GlobalPullSecret needs to be on HiveConfig, otherwise it would require the caller to set it on every ClusterDeployment. The goal here is for the cluster admin to be able to configure a pull secret to be used globally or in addition to whatever the user sets. (depending on the release image they use and what registry it's located on)

Right, my mistake. Fixing it